wave_vision/tests/detector_tests/orb_tests.cpp

```diff
@@ -0,0 +1,137 @@
+TEST(ORBDetectorTests, GoodConfig) {
+    // Default
+    EXPECT_NO_THROW(ORBDetectorParams config1);
+
+    // Custom params struct (with good values)
+    int num_features = 500;
+    float scale_factor = 1.2f;
+    int num_levels = 8;
+    int edge_threshold = 31;
+    int score_type = cv::ORB::HARRIS_SCORE;
+    int fast_threshold = 20;
+
+    EXPECT_NO_THROW(ORBDetectorParams config2(num_features,
+                                              scale_factor,
+                                              num_levels,
+                                              edge_threshold,
+                                              score_type,
+                                              fast_threshold));
+
+    // From orb.yaml file, with good values.
+    EXPECT_NO_THROW(ORBDetectorParams config3(TEST_CONFIG););
+}
+
+// Checks that an incorrect configuration path throws an exception
+TEST(ORBDetectorTests, BadConfigPath) {
+    const std::string bad_path = "bad_path";
+
+    ASSERT_THROW(ORBDetectorParams bad_config(bad_path), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, ConstructorTest) {
+    ORBDetectorParams config;
+
+    EXPECT_NO_THROW(ORBDetector detector);
+    EXPECT_NO_THROW(ORBDetector detector1(config));
+}
+
+// Check that incorrect parameter values throw exceptions.
+TEST(ORBDetectorTests, BadNumFeatures) {
+    ORBDetectorParams config;
+    config.num_features = -1;
+
+    ASSERT_THROW(ORBDetector bad_nfeatures(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, BadScaleFactor) {
+    ORBDetectorParams config;
+    config.scale_factor = 0.99;
+
+    ASSERT_THROW(ORBDetector bad_scalefactor1(config), std::invalid_argument);
+
+    config.scale_factor = -1;
+    ASSERT_THROW(ORBDetector bad_scalefactor2(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, BadNumLevels) {
+    ORBDetectorParams config;
+    config.num_levels = 0;
+
+    ASSERT_THROW(ORBDetector bad_nlevels1(config), std::invalid_argument);
+
+    config.num_levels = -1;
+    ASSERT_THROW(ORBDetector bad_nlevels2(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, BadEdgeThreshold) {
+    ORBDetectorParams config;
+
+    config.edge_threshold = -1;
+
+    ASSERT_THROW(ORBDetector bad_edgethreshold(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, BadScoreType) {
+    ORBDetectorParams config;
+
+    config.score_type = -1;
+
+    ASSERT_THROW(ORBDetector bad_scoretype1(config), std::invalid_argument);
+
+    config.score_type = 2;
+    ASSERT_THROW(ORBDetector bad_scoretype2(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, BadFastThreshold) {
+    ORBDetectorParams config;
+    config.fast_threshold = 0;
+
+    ASSERT_THROW(ORBDetector bad_fastthreshold(config), std::invalid_argument);
+
+    config.fast_threshold = -1;
+    ASSERT_THROW(ORBDetector bad_fastthreshold1(config), std::invalid_argument);
+}
+
+TEST(ORBDetectorTests, ConfigurationTests) {
```

Comment:
These are some very detailed tests of the configuration functionality, :hushed: :+1: